src/TemplateProcessor.ts

```diff
@@ -1090,6 +1099,64 @@ export default class TemplateProcessor {
         return null;
     }
 
+
+    async waitCondition(waitCondition, timeout = 10000) {
+        return new Promise((resolve, reject) => {
+            const startTime = Date.now();
+            let conditionExpression;
+            let timeoutId;  // Declare timeoutId here
+
+            try {
+                conditionExpression = jsonata(waitCondition);
+            } catch (e) {
+                this.logger.error(`invalid wait condition expression: ${waitCondition}, ${e}`);
+                resolve({
+                    "error": {
+                        message: e.message,
+                        name: e.name,
+                        stack: e.stack,
+                        input: this.input
+                    }
+                });
+            }
+
+            const checkTimeout = () => {
+                if (Date.now() - startTime >= timeout) {
+                    this.removeDataChangeCallback('/');
+                    this.logger.debug(`wait condition ${waitCondition} timed out in ${timeout}ms`);
+                    resolve({
+                        "error": {
+                            message: `wait condition ${waitCondition} timed out in ${timeout}ms`,
+                            output: this.output
+                        }
+                    });
+                } else {
+                    timeoutId = setTimeout(checkTimeout, 100);  // Assign the timeout ID here
+                }
+            };
+
+            let checkConditionCallback = async (data, jsonPtr) => {
+                let matchedCondition = await conditionExpression.evaluate(this.output);
+                if (matchedCondition) {
```

Comment:
this should probably be if(matchedCondition===true) unless you really mean to allow conditions like "42" which will be truthy